### ISSUE TITLE
Security: Build from new Image

### ISSUE THREAD

**Issue by genofire** [gramps-project/Gramps.js#121]
I got an security report from trivy (aquasec) with a lot of vulnibilities - mainly in the basis-images:

```diff
Report:
  Artifact:
-   Repository:  dmstraub/gramps
-   Tag:         latest
+   Repository:  gramps-project/grampsweb
+   Tag:         v0.16.1
  Registry:
-   Server:  index.docker.io
+   Server:  ghcr.io
  Scanner:
    Name:     Trivy
    Vendor:   Aqua Security
    Version:  0.31.3
  Summary:
    Critical Count:  55
-   High Count:      107
+   High Count:      98
    Low Count:       17
-   Medium Count:    112
+   Medium Count:    110
    None Count:      0
    Unknown Count:   3
  ```

please rebuild the image layers on new basis-images.

(I do not put all 283 CVEs hier inside ....)

---
image digest:

- docker.io/dmstraub/gramps@sha256:a24d287e4d57011cd436ce636a04aa1ed1b57cb77aba7f1f6cfb951a7e947695
- ghcr.io/gramps-project/grampsweb@sha256:b5b3c5e39a8a94db59157846989ff63640b4d4dd6b5080c14e70345543198954

**Comment by DavidMStraub** [comment]
Can you please be more specific?

Here are the docker files:
https://github.com/DavidMStraub/gramps-docker/blob/master/Dockerfile
https://github.com/gramps-project/gramps-webapi/blob/master/Dockerfile
https://github.com/gramps-project/Gramps.js/blob/main/Dockerfile

What would you like to be updated?

**Comment by genofire** [comment]
every layer -> base image is an 8 month old `debian:bullseye` ...

the most vulnerabilities are in there.

(Does you do not install updates on your server for over 8 month?)

**Comment by DavidMStraub** [comment]
> (Does you do not install updates on your server for over 8 month?)

Please stay polite or I will lock this thread.

**Comment by genofire** [comment]
Sorry, i do not want to be rough, just to explain the comparison.

Here are some CRITICAL-Reported CVEs:

- https://avd.aquasec.com/nvd/2022/cve-2022-1664/
- https://avd.aquasec.com/nvd/cve-2021-33574
- https://avd.aquasec.com/nvd/cve-2022-23218
- https://avd.aquasec.com/nvd/cve-2022-23219
- https://avd.aquasec.com/nvd/cve-2021-33574
- https://avd.aquasec.com/nvd/cve-2022-23218
- https://avd.aquasec.com/nvd/cve-2022-23219

**Comment by DavidMStraub** [comment]
Thanks. It would be great if you could scan the most recent build of debian:bullseye. I wonder if that is much better.

**Comment by genofire** [comment]
My Reports just shows fixed CVEs:
```
Report:
  Artifact:
    Repository:  library/debian
    Tag:         bullseye
  Registry:
    Server:  index.docker.io
  Scanner:
    Name:     Trivy
    Vendor:   Aqua Security
    Version:  0.31.3
  Summary:
    Critical Count:  0
    High Count:      0
    Low Count:       0
    Medium Count:    0
    None Count:      0
    Unknown Count:   0
  Update Timestamp:  2022-10-16T19:29:04Z
  Vulnerabilities:
  ```

**Comment by DavidMStraub** [comment]
Thanks! For webapi, I think it also makes sense, while here at the moment there should be enough commit activity...

**Comment by genofire** [comment]
With https://github.com/gramps-project/Gramps.js/commit/0526ab261caaa124b0fc492c92ae1a2c4e73851a there was build an new image, with down till a new bullseye-image.

That current security scan looks good.
Current state: all CVEs are in libc - just new findings in the bullseye - so nothing importend yet.
Also nothing in any gramps related dependencies (like python or python libraries)

Thanks for your support.

Here my full Report
```
Report:
  Artifact:
    Repository:  gramps-project/grampsweb
    Tag:         latest-devel
  Registry:
    Server:  ghcr.io
  Scanner:
    Name:     Trivy
    Vendor:   Aqua Security
    Version:  0.31.3
  Summary:
    Critical Count:  0
    High Count:      7
    Low Count:       0
    Medium Count:    11
    None Count:      0
    Unknown Count:   0
  Update Timestamp:  2022-10-21T21:59:19Z
  Vulnerabilities:
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2021-4037
    Resource:           linux-libc-dev
    Score:              4.4
    Severity:           HIGH
    Target:
    Title:              kernel: security regression for CVE-2018-13405
    Vulnerability ID:   CVE-2021-4037
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-20421
    Resource:           linux-libc-dev
    Score:              7.8
    Severity:           HIGH
    Target:
    Title:              In binder_inc_ref_for_node of binder.c, there is a possible way to cor ...
    Vulnerability ID:   CVE-2022-20421
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-2602
    Resource:           linux-libc-dev
    Severity:           HIGH
    Target:
    Title:              [io_uring/af_unix: defer registered files gc to io_uring release]
    Vulnerability ID:   CVE-2022-2602
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-3176
    Resource:           linux-libc-dev
    Score:              7
    Severity:           HIGH
    Target:
    Title:              kernel: use-after-free in io_uring for POLLFREE notification with Signalfd_poll() and binder_poll()
    Vulnerability ID:   CVE-2022-3176
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-41674
    Resource:           linux-libc-dev
    Score:              7.3
    Severity:           HIGH
    Target:
    Title:              kernel: u8 overflow problem in  cfg80211_update_notlisted_nontrans()
    Vulnerability ID:   CVE-2022-41674
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-42719
    Resource:           linux-libc-dev
    Score:              7.3
    Severity:           HIGH
    Target:
    Title:              kernel: A use-after-free problem observed in multi-BSSID element when parsing
    Vulnerability ID:   CVE-2022-42719
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-42720
    Resource:           linux-libc-dev
    Score:              7.3
    Severity:           HIGH
    Target:
    Title:              kernel: A use-after-free problem was observed in bss_ref_get in net/wireless/scan.c
    Vulnerability ID:   CVE-2022-42720
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-0171
    Resource:           linux-libc-dev
    Score:              4.7
    Severity:           MEDIUM
    Target:
    Title:              kernel: KVM: cache incoherence issue in SEV API may lead to kernel crash
    Vulnerability ID:   CVE-2022-0171
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-1184
    Resource:           linux-libc-dev
    Score:              5.5
    Severity:           MEDIUM
    Target:
    Title:              kernel: use-after-free and memory errors in ext4 when mounting and operating on a corrupted image
    Vulnerability ID:   CVE-2022-1184
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-2663
    Resource:           linux-libc-dev
    Score:              5.9
    Severity:           MEDIUM
    Target:
    Title:              kernel/netfilter: nf_conntrack_irc message handling issue
    Vulnerability ID:   CVE-2022-2663
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-3061
    Resource:           linux-libc-dev
    Score:              6.2
    Severity:           MEDIUM
    Target:
    Title:              kernel: fbdev: i740fb: divide-by-zero in drivers/video/fbdev/i740fb.c could lead to DoS
    Vulnerability ID:   CVE-2022-3061
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-3303
    Resource:           linux-libc-dev
    Score:              4.4
    Severity:           MEDIUM
    Target:
    Title:              kernel: race condition in snd_pcm_oss_sync leads to NULL pointer dereference
    Vulnerability ID:   CVE-2022-3303
    Fixed Version:      5.10.148-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-3586
    Resource:           linux-libc-dev
    Score:              6.1
    Severity:           MEDIUM
    Target:
    Title:              kernel: net scheduler use-after-free information disclosure vulnerability
    Vulnerability ID:   CVE-2022-3586
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-39188
    Resource:           linux-libc-dev
    Score:              4.7
    Severity:           MEDIUM
    Target:
    Title:              Kernel: unmap_mapping_range() race with munmap() on VM_PFNMAP mappings leads to stale TLB entry
    Vulnerability ID:   CVE-2022-39188
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-39842
    Resource:           linux-libc-dev
    Score:              6.1
    Severity:           MEDIUM
    Target:
    Title:              Kernel: A type conflict of size_t versus int cause an integer overflow in pxa3xx_gcu_write
    Vulnerability ID:   CVE-2022-39842
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-40307
    Resource:           linux-libc-dev
    Score:              4.7
    Severity:           MEDIUM
    Target:
    Title:              kernel: use-after-free in efi_capsule_write in capsule-loader.c
    Vulnerability ID:   CVE-2022-40307
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:       https://avd.aquasec.com/nvd/cve-2022-42721
    Resource:           linux-libc-dev
    Score:              5.7
    Severity:           MEDIUM
    Target:
    Title:              kernel: BSS list corruption in cfg80211_add_nontrans_list in net/wireless/scan.c
    Vulnerability ID:   CVE-2022-42721
    Fixed Version:      5.10.149-1
    Installed Version:  5.10.140-1
    Links:
    Primary Link:      https://avd.aquasec.com/nvd/cve-2022-42722
    Resource:          linux-libc-dev
    Score:             5.5
    Severity:          MEDIUM
    Target:
    Title:             Kernel: Denial of service in beacon protection for  P2P-device
    Vulnerability ID:  CVE-2022-42722
```
